R/get_fs_int.R

```diff
@@ -0,0 +1,62 @@
+#' Compute interaction indicators for tspa() function
+#' 
+#' @param dat A data frame containing first-order factor score indiactors.
+#' @param fs_name A character vector indicating names of factor scores variables.
+#' @param se A vector indicating standard error of factor scores
+#' @param loading A vector indicating model-implied loadings of factor scores
+#' @param model An optional string specifying the measurement model
+#'              in \code{lavaan} syntax.
+#' @return A data frame of product indicators for interaction terms,
+#'         with their loadings and standard errors.
+#'
+#' @importFrom utils combn
+#'
+#' @export
+
+get_fs_int <- function(dat, fs_name, se, loading, model = NULL) {
+
+  # Connect fs and se
+  fs_list <- mapply(function(x, y, z) list(name = x, se = y, loading = z),
+                    fs_name, se, loading, SIMPLIFY = FALSE)
+
+  # Create fs pairs
+  if (is.null(model)) {
+    fs_pairs <- combn(fs_name, 2, simplify = FALSE)
+  } else {
+    elements <- trimws(unlist(strsplit(model, "\\+")))
+    fs_pairs <- lapply(strsplit(elements, ":"), function(pair) {
+      sapply(pair, function(element) {
+        paste0(names(element), element)
+      })})
+  }
+
+  # Observation-specific check
+  check_element <- function(x) {
+    if(var(x, na.rm = TRUE) == 0) {
+      return(unlist(x[1]))
+    } else {
+      return(x)
+    }
+  }
```

Comment:
Not needed with the change in input type

```suggestion

```

Reply:
Removed